seq2science/rules/get_genome.smk

```diff
@@ -118,3 +127,31 @@ rule decoy_transcripts:
     shell:
          ("cpulimit --include-children -l {threads}00 -- " if config.get("cpulimit", True) else " ") +
          "sh {params.script} -j {threads} -g {input.genome} -a {input.gtf} -t {input.transcripts} -o $(dirname {output}) > {log} 2>&1"
+
+
+@lru_cache(maxsize=None)
+def has_annotation(assembly):
+    """
+    returns True/False on whether or not the assembly has an annotation.
+    """
+    # check if genome providd by user or already downloaded, if so check if the annotation came along
+    if all(os.path.exists(f"{config['genome_dir']}/{assembly}.{extension}") for extension in config["genome_types"]):
+        return os.path.exists(f"{config['genome_dir']}/{assembly}.annotation.gtf")
+
+    if "provider" in config:
+        providers = [config["provider"]]
+    else:
+        providers = ["Ensembl", "UCSC", "NCBI"]
+
+    # check if we expect an annotation
+    # we do not want genomepy outputting to us that it is downloading stuff
+    with open(os.devnull, 'w') as null:
+        with contextlib.redirect_stdout(null), contextlib.redirect_stderr(null):
+            for provider in providers:
+                p = genomepy.ProviderBase.create(provider)
+                if assembly in p.genomes and \
+                    p.get_genome_download_link(assembly) is not None:
```

Comment:
isn't this double? If the assembly is in p.genomes, there should always be a genome_download_link.

Reply:
I guess? I don't know how genomepy works internally :). I had some issues with just using get_genome_download_link, and added the check if its in genomes. I didn't make the effort to check if it's superfluous. :angel: